ReactApp/src/misc/api-fetch-setup.js

```diff
@@ -3,26 +3,24 @@
  */
 import apiFetch from '@wordpress/api-fetch';
 
+/**
+ * Internal dependencies
+ */
+import { getGBKit } from './store';
+
 /**
  * Initializes the API fetch configuration and middleware.
  *
  * This function sets up the root URL middleware, adds headers to requests,
  * and preloads some endpoints with mock data for specific components.
  */
 export function initializeApiFetch() {
-	apiFetch.use(apiFetch.createRootURLMiddleware(''));
+	const { siteURL, authToken } = getGBKit();
+	const rootURL = siteURL ? `${siteURL}/wp-json/` : undefined;
 
-	apiFetch.use((options, next) => {
-		options.headers = options.headers || new Headers();
-		options.headers.set('Content-Type', 'application/json');
```

Comment:
I found that using the `Headers` interface and `Headers.prototype.set` resulted in the values never being present in the request headers. It appears that `apiFetch` [expects an object](https://github.com/WordPress/gutenberg/blob/5dae25b15e82a59d28c6c065b5a3745f3f83980f/packages/api-fetch/src/index.js#L93-L94) instead, so I refactored this to an object.

Reply:
Additionally, the `Content-Type` request header is [managed by](https://github.com/WordPress/gutenberg/blob/5dae25b15e82a59d28c6c065b5a3745f3f83980f/packages/api-fetch/src/index.js#L96-L100) `@wordpress/api-fetch` as long as we use the default fetcher, so I removed this redundant value.